File: 209. Minimum Size Subarray Sum/solution.cpp
#include <algorithm>
#include <cstdint>
#include <vector>

using namespace std;

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {

      int sum = 0, start = 0, end = 0, ans = INT32_MAX, len = nums.size();

      while (end < len) {
        sum += nums[end];
        while (sum >= target) {
          ans = min(ans, (end - start) + 1);
          sum -= nums[start];
          start++;
        }

        end++;
      }

      return ans == INT32_MAX ? 0 : ans; 
    }
};

File: 209. Minimum Size Subarray Sum/solution.py
import sys
from typing import List

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        start = 0 
        end = 0 
        ans = sys.maxsize 
        sum = 0

        while end < len(nums): 
            sum += nums[end]
            if sum >= target: 
                ans = min(ans, end - start)
                start += 1
            end += 1


        return ans if ans != sys.maxsize else 0
        
